src/steps/project/index.ts

```diff
@@ -0,0 +1,107 @@
+
+import { createAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+
+import { Entities, Relationships, Steps } from '../constants';
+import { createProjectEntity } from './converter';
+
+export async function fetchProjects({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(
+    { _type: Entities.CONFIGURATION._type },
+    async (pipelineEntity) => {
+      const pipelineDetail = await apiClient.fetchPipelineDetails(
+        pipelineEntity.id as string,
+      );
+
+      const project = await apiClient.fetchProjectDetails(
+        pipelineDetail.project_slug,
+      );
+
+      const isProjectEntity = await jobState.findEntity(
+        `circleci_project:${project.id}`,
+      );
+
+      if (!isProjectEntity) {
+        const projectEntity = createProjectEntity(project);
+        await jobState.addEntity(projectEntity);
+
+        await jobState.addRelationship(
+          createDirectRelationship({
+            _class: RelationshipClass.HAS,
+            from: projectEntity,
+            to: pipelineEntity,
+          }),
+        );
+      } else {
+        const projectEntity = isProjectEntity;
+
+        await jobState.addRelationship(
+          createDirectRelationship({
+            _class: RelationshipClass.HAS,
+            from: projectEntity,
+            to: pipelineEntity,
+          }),
+        );
+      }
+    },
+  );
+}
+
+export async function buildUserGroupAndProjectRelationship({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+  await jobState.iterateEntities(
+    { _type: Entities.REPOSITORY._type },
+    async (repositoryEntity) => {
+      const repositoryDetail = await apiClient.fetchProjectDetails(
+        repositoryEntity.id as string,
+      );
+
+      const userGroupEntity = await jobState.findEntity(
+        `circleci_user_group:${repositoryDetail.organization_id.toString()}`,
+      );
+      if (userGroupEntity) {
+        await jobState.addRelationship(
+          createDirectRelationship({
+            _class: RelationshipClass.HAS,
+            from: userGroupEntity,
+            to: repositoryEntity,
+          }),
+        );
+      }
+    },
+  );
+}
+
+export const projectsSteps: IntegrationStep<IntegrationConfig>[] = [
+  {
+    id: Steps.PROJECTS,
+    name: 'Fetch Projects',
+    entities: [Entities.REPOSITORY],
+    relationships: [Relationships.PROJECT_HAS_PIPELINE],
+    dependsOn: [Steps.PIPELINES, Steps.USER_GROUPS],
+    executionHandler: fetchProjects,
+  },
+
+  {
+    id: Steps.BUILD_USER_GROUPS_PROJECTS_RELATIONSHIPS,
+    name: 'Build Project and Pipelines Relationships',
+    entities: [],
+    relationships: [Relationships.USER_GROUP_HAS_PROJECT],
+    dependsOn: [Steps.USER_GROUPS, Steps.PIPELINES, Steps.PROJECTS],
```

Comment:
This seems to be unnecessary: Steps.PIPELINES

```suggestion
    dependsOn: [Steps.USER_GROUPS, Steps.PROJECTS],
```